src/CacheTagsHash.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Drupal\fastly;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+
+/**
+ * Class CacheTagsHash.
+ *
+ * @package Drupal\fastly
+ */
+class CacheTagsHash implements CacheTagsHashInterface {
+
+  /**
+   * ConfigFactory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Fastly settings.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  protected $config;
+
+  /**
+   * CacheTagsHash constructor.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   */
+  public function __construct(ConfigFactoryInterface $config_factory) {
+    $this->configFactory = $config_factory;
+    $this->config = $config_factory->get('fastly.settings');
+  }
+
+  /**
+   * Maps cache tags to hashes.
+   *
+   * Used when the Surrogate-Key/X-Drupal-Cache-Tags header size otherwise
+   * exceeds 16 KB.
+   *
+   * @param string[] $cache_tags
+   *   The cache tags in the header.
+   *
+   * @return string[]
+   *   The hashes to use instead in the header.
+   */
+  public function cacheTagsToHashes(array $cache_tags) {
+    $hashes = [];
+    $siteId = getenv('FASTLY_SITE_ID') ?: $this->config->get('site_id');
+    if (!$siteId) {
+      $siteId = substr(md5(microtime()), 0, 7);
```

Comment:
I would just use the built in Drupal function https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Utility%21Random.php/function/Random%3A%3Astring/8.2.x, this seems more readable.

Reply:
Modified to suggested.